internal/publish/publish.go

```diff
@@ -0,0 +1,180 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package publish
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/google/go-github/v32/github"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/builder"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/storage"
+)
+
+const (
+	snapshotStage   = "snapshot"
+	stagingStage    = "staging"
+	productionStage = "production"
+)
+
+// Package function publishes the current package to the package-storage.
+func Package(githubUser string, githubClient *github.Client, skipPullRequest bool) error {
+	packageRoot, err := packages.MustFindPackageRoot()
+	if err != nil {
+		return errors.Wrap(err, "locating package root failed")
+	}
+
+	m, err := packages.ReadPackageManifestFromPackageRoot(packageRoot)
+	if err != nil {
+		return errors.Wrapf(err, "reading package manifest failed (path: %s)", packageRoot)
+	}
+
+	buildDir, found, err := builder.FindBuildPackagesDirectory()
+	if err != nil {
+		return errors.Wrap(err, "locating build directory failed")
+	}
+	if !found {
+		return errors.New("build directory not found. Please run 'elastic-package build' first")
+	}
+
+	builtPackageDir := filepath.Join(buildDir, m.Name, m.Version)
+	fmt.Printf("Use build directory: %s\n", builtPackageDir)
+	_, err = os.Stat(builtPackageDir)
+	if os.IsNotExist(err) {
+		return errors.Wrapf(err, "build directory '%s' is missing. Please run 'elastic-package build' first", builtPackageDir)
+	}
+	if err != nil {
+		return errors.Wrapf(err, "stat file failed: %s", builtPackageDir)
+	}
+
+	fmt.Println("Clone package-storage repository")
+	r, err := storage.CloneRepository(githubUser, productionStage)
+
+	fmt.Printf("Find latest package revision of \"%s\" in package-storage\n", m.Name)
+	latestRevision, stage, err := findLatestPackageRevision(r, m.Name)
+	if err != nil {
+		return errors.Wrap(err, "can't find latest package revision")
+	}
+
+	if latestRevision == nil {
+		fmt.Println("Package has not been published yet")
+	} else if latestRevision.Version == m.Version {
+		fmt.Println("Package has already been published")
```

Comment:
Maybe include the package version here and also the stage where the already-published package version can be found, in case the user wants to do some inspection?

Reply:
Fixed